### ISSUE TITLE
Accuracy of Explaination in SHAP

### ISSUE THREAD

**Issue by LividWo** [shap/shap#222]
I tried the notebook: League of Legends Win Prediction with XGBoost
And have the following ovservation:
For some cases, although the classifier could correctly predict the label (win for instance), SHAP seems made a negative explaination (output value < base value, or i should say wrong prediction?).
In my understanding, when output value lower than base value, the sample is regard as lose the game.

**My question:**
**# Is this inconsistence between prediction and explaination occurs because shap make a wrong explaination?** 
Any plan of adding metrics to measure SHAP's accuracy?

**Comment by slundberg** [comment]
Could you be more specific about the problem? For example give some values for a specific sample from the dataset? The sum of the shap_values + explainer.expected_value should exactly equal the log odds output of the model for XGBoost.

I am working on metrics, but the problem you are discussing shouldn't exist, hence wouldn't show up unless there is a bug.

**Comment by LividWo** [comment]
![image](https://user-images.githubusercontent.com/12826343/44194275-0f4a5300-a168-11e8-99db-aeca8efdb2a3.png)

In the example above
The output value 1.48 equals ~96% in probability, so XGBoost regards this sample as more likely to be 1, which is correct according to the ground truth. However, from the figure, it seems like more features are pushing the prediction lower. And this is a bit misleading, before your reply, I regard a sample as 0 when output value < base value. Now I understand which is not the case.

My question is:
1. How should I interpret the relationship between the output value and base value? 
2. What conclusion can we draw when the output value is lower than the base value?
3. Minor question about xgboost. Given a probability, xgboost need a threshold to do binary classification, how can we get this threshold?

**Comment by slundberg** [comment]
1. If your base value is 2.15 then you must have a very high base rate for the model. In other words, the model was trained on data that was almost all 1's. If you have no information about a specific example you will just predict 2.15, but as you learn more and more about a specific sample the values of the features for that sample impact your prediction. In this case they lower the probability, but not by enough to make 0 more likely than 1 (since you started very biased towards 1).
2. It means that the current sample has a lower output from the model than the typical sample. So the features for this sample are lowering the model output from what you would expect on average.
3. I am not sure, but if you are using the sklearn predict API I bet they use 0.5 probability which is 0 log-odds.

....hope that is helpful!

**Comment by lingeshsuba** [comment]
> ![image](https://user-images.githubusercontent.com/12826343/44194275-0f4a5300-a168-11e8-99db-aeca8efdb2a3.png)
> 
> In the example above
> The output value 1.48 equals ~96% in probability, so XGBoost regards this sample as more likely to be 1, which is correct according to the ground truth. However, from the figure, it seems like more features are pushing the prediction lower. And this is a bit misleading, before your reply, I regard a sample as 0 when output value < base value. Now I understand which is not the case.
> 
> My question is:
> 
> 1. �How should I interpret the relationship between the output value and base value?
> 2. What conclusion can we draw when the output value is lower than the base value?
> 3. Minor question about xgboost. Given a probability, xgboost need a threshold to do binary classification, how can we get this threshold?

sir,have you got any other answer for this question
